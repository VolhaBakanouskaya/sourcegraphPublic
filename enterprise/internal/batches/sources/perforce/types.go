package perforce

import "github.com/sourcegraph/sourcegraph/internal/extsvc/perforce"

// from @varsanojidan, in a comment on the PR:
// You only really need this if the Changelist model doesn't contain information about changelist builds or reviews
// (if perforce has such a concept), if you can get all the info from Changelist directly,
// then you can just use the Changelist type instead.
//
// at some point we will probably want to add info about CLs that we would get from Swarm or elsewhere
// so I'll leave this in here
type AnnotatedChangelist struct {
	*perforce.Changelist
}
