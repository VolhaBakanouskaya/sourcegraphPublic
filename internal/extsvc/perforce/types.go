package perforce

import (
	"time"
)

// TODO: this was just blindly copied from the ADO implementation in order to make things "work" initially
// need to probably switch most of this out for Changelist vernacular

type ChangelistStatus string

type CreatorInfo struct {
	ID          string `json:"id"`
	DisplayName string `json:"displayName"`
	UniqueName  string `json:"uniqueName"`
	URL         string `json:"url"`
	ImageURL    string `json:"imageUrl"`
}

type Changelist struct {
	Depot        Depot            `json:"depot"`
	ID           int              `json:"changelistId"`
	Status       ChangelistStatus `json:"status"` // pending, submitted, shelved
	CreationDate time.Time        `json:"creationDate"`
	CreatedBy    CreatorInfo      `json:"createdBy"`
	Title        string           `json:"title"`
	Description  string           `json:"description"`
	URL          string           `json:"url"`
}

type Repository struct {
	ID         string `json:"id"`
	Name       string `json:"name"`
	CloneURL   string `json:"remoteURL"`
	IsDisabled bool   `json:"isDisabled"`
}
