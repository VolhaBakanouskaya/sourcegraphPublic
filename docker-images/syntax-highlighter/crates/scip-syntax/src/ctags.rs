use scip::types::Descriptor;
use serde::{Deserialize, Serialize};

#[derive(Debug)]
pub enum TagKind {
    Function,
    Class,
}

#[derive(Debug)]
pub struct TagEntry {
    pub descriptors: Vec<Descriptor>,
    pub kind: TagKind,
    pub parent: Option<Box<TagEntry>>,

    pub line: usize,
    // pub column: usize,
}

#[derive(Serialize, Deserialize, Debug)]
#[serde(tag = "command")]
pub enum Request {
    #[serde(rename = "generate-tags")]
    GenerateTags {
        // command == generate-tags
        filename: String,
        size: usize,
    },
}

#[derive(Serialize, Deserialize, Debug)]
#[serde(tag = "_type")]
pub enum Reply {
    #[serde(rename = "program")]
    Program { name: String, version: String },
    #[serde(rename = "completed")]
    Completed { command: String },
    #[serde(rename = "error")]
    Error { message: String, fatal: bool },
    #[serde(rename = "tag")]
    Tag {
        name: String,
        path: String,
        language: String,
        /// Starts at 1
        line: usize,
        kind: String,
        pattern: String,
        scope: Option<String>,
        #[serde(rename = "scopeKind")]
        scope_kind: Option<String>,
        signature: Option<String>,
        // TODO(SuperAuguste): Any other properties required? Roles? Access?
    },
}
