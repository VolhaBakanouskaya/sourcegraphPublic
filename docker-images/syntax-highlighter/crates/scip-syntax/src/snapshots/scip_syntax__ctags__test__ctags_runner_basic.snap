---
source: crates/scip-syntax/src/ctags.rs
expression: "String::from_utf8(output.get_ref().to_vec())?"
---
{"_type":"program","name":"SCIP Ctags","version":"5.9.0"}
{"_type":"tag","name":"other","path":"main.rs","language":"go","line":6,"kind":"method","scope":null}
{"_type":"tag","name":"something","path":"main.rs","language":"go","line":5,"kind":"method","scope":null}
{"_type":"tag","name":"main","path":"main.rs","language":"go","line":1,"kind":"method","scope":null}
{"_type":"completed","command":"generate-tags"}

